### PR TITLE
chore(updatecli) Fix JDK manifests to use the proper Ubuntu version

### PR DESCRIPTION
**Pull request by gounthar**
Moved from `focal` to `jammy` for JDK17/21 and added the architecture for Windows and `linux/arm/v7` docker images.
JDK11 will fail because of :

> [jdk-11.0.22+7.1](https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.22%2B7.1) contains a respin for aarch64_mac described in the 'aarch64 macOS Respin' section of this blog post https://adoptium.net/blog/2024/01/eclipse-temurin-8u402-11022-1710-and-2102-available/
> So since it was specific to that platform, it is not relevant for any containers we currently publish to Docker Hub.

### Testing done

`updatecli diff --config ./updatecli/updatecli.d/jdk21.yaml --values ./updatecli/values.github-action.yaml`
`updatecli diff --config ./updatecli/updatecli.d/jdk17.yaml --values ./updatecli/values.github-action.yaml`
`updatecli diff --config ./updatecli/updatecli.d/jdk11.yaml --values ./updatecli/values.github-action.yaml`

### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- ~~Link to relevant pull requests, esp. upstream and downstream changes~~
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: updatecli/updatecli.d/jdk21.yaml [hunk 1]
```diff
@@ -51,44 +51,44 @@ conditions:
       tag: '{{ source "getLatestJDK21Version" }}-jdk-alpine'
   checkTemurinDebianDockerImages:
     kind: dockerimage
-    name: Check if the container image "eclipse-temurin:<lastVersion>-jdk-focal" is available
+    name: Check if the container image "eclipse-temurin:<lastVersion>-jdk-jammy" is available
     disablesourceinput: true
     spec:
       architectures:
         - amd64
```
[the remark this answers is not in the document]
Comment (gounthar) [reply updatecli/updatecli.d/jdk21.yaml:58]: No, as in [the documentation](https://www.updatecli.io/docs/plugins/resource/docker_image/), I can read:

> The default operating system is linux.

[the remark this answers is not in the document]
Comment (gounthar) [reply updatecli/updatecli.d/jdk21.yaml:58]: Yes, doesn't hurt, it's some kind of "the documentation is the code". 👍

**Comment by gounthar** [comment]
Got a 503, as the #761.

**Comment by gounthar** [comment]
Niw it builds! 🚀

**Comment by MarkEWaite** [comment]
@gounthar can you resolve the conflicts in this that came from my merge of

* #761

**Comment by gounthar** [comment]
It's done.
Our two PRs were quite similar, except Tim went way farther with rewriting the script. 😉 
Thanks, Mark. 👍

**Review by dduportal — APPROVED** [review]
```
Run Summary
===========
Pipeline(s) run:
  * Changed:	0
  * Failed:	0
  * Skipped:	0
  * Succeeded:	10
  * Total:	10
```

LGTM!


Food for though (for another PR): we should be able to add a 2nd source to updatecli manifest which would retrieve the current Docker image suffix (or distribution) to make it more universal. But it is nitpicking.

**Pull request merged**